### PR TITLE
upgrade cloud watch

### PR DESCRIPTION
**Pull request by dgzpg**
# Description
upgrade cloud watch

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageManager.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        synchronized (lock) {
+            if (this.shutdownInProgress) {
+                return;
+            }
+            this.shutdownInProgress = true;
+        }
+        sender.shutdown();
+    }
+
+    class ClientFinalizer extends Thread {
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageManager.java:114]: thread name

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * 消息发送者，监听日志队列并选择具体日志收集者进行日志输出
+ */
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java:41]: don't chinese

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    private final ChunkQueue chunkQueue;
+    private final LinkedList<CompressedChunk> readCompressedChunk;
+    private volatile boolean running = true;
+    private volatile CountDownLatch shutdownLatch;
+
+    public MessageSender(ChunkQueue chunkQueue) {
+        this.chunkQueue = chunkQueue;
+        this.readCompressedChunk = new LinkedList<>();
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java:67]: thread name

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * 从队列中拉出数据并且组合成一个list进行发送
+     */
+    @Override
+    public void run() {
+        while (running) {
+            try {
+                //从队列中取数据
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageSender.java:87]: dont chinese

**Review by kevinten10 — COMMENTED** [review]
review

File: capa-spi-aws-log/pom.xml [hunk 5]
```diff
@@ -33,9 +33,16 @@
...
+        <capa.addons.version>1.0.4.RELEASE</capa.addons.version>
+        <sentinel.core.version>1.8.2</sentinel.core.version>
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.22</version>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (kevinten10) [reply capa-spi-aws-log/pom.xml:46]: please move test denpendency to the end

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageManager.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        //TODO 设置队列最大值
+        chunkQueueMaxBytes = defaultChunkQueueMaxBytes;
+        chunkQueue = new ChunkQueue(chunkQueueMaxBytes);
+        sender = new MessageSender(chunkQueue);
+        startNewSender();
+        consumer = createConsumer();
+    }
+
+    public static MessageManager getInstance() {
+        synchronized (lock) {
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/handle/MessageManager.java:74]: 使用双重检查锁

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java [hunk 7]
```diff
@@ -117,35 +130,95 @@ public static void putLogEvent(String message) {
...
+                    longCounter.bind(Attributes.of(AttributeKey.stringKey(CLOUD_WATCH_LOGS_PUT_LOG_EVENTS_ERROR_TYPE), statusCode))
+                            .add(COUNTER_NUM);
+                });
+            } finally {
+            }
+        }
+    }
+
+    public static List<String> getLogStreamNames() {
+        String ip = Foundation.net().getHostAddress() == null
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java:183]: this is trip logic, not aws logic.
能做成插件化的么

Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java:183]: this is trip logic, not aws logic.
能做成插件化的么

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java [hunk 8]
```diff
@@ -117,35 +130,95 @@ public static void putLogEvent(String message) {
...
+        return logStreamNames;
+    }
+
     private static String buildAppId() {
         return CapaFoundation.getAppId(FoundationType.TRIP);
     }
 
     private static String buildApplicationEnv() {
-        return String.format(APPLICATION_ENV_FORMAT, CapaFoundation.getEnv(FoundationType.TRIP));
+        return CapaFoundation.getEnv(FoundationType.TRIP) == "" ? "DEFAULT" : CapaFoundation.getEnv(FoundationType.TRIP);
```
Comment (kevinten10) [inline capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java:198]: this is trip logic, not aws logic.
能做成插件化的么

**Review by kevinten10 — COMMENTED** [review]
review

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java (hunk 7 above)
Comment (dgzpg) [reply capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java:183]: 后续可以做成插件化，现在是临时使用

File: capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java (hunk 8 above)
Comment (dgzpg) [reply capa-spi-aws-log/src/main/java/group/rxcloud/capa/spi/aws/log/service/CloudWatchLogsService.java:198]: 现在由于capafoundation没有提供完整的方法，所以这里临时兜底下。

File: capa-spi-aws-log/pom.xml [hunk 9]
```diff
@@ -33,14 +33,24 @@
         <log4j.version>2.14.1</log4j.version>
         <logback.version>1.2.3</logback.version>
         <gson.version>2.8.0</gson.version>
+        <capa.addons.version>1.0.5.RELEASE</capa.addons.version>
```
Comment (kevinten10) [inline capa-spi-aws-log/pom.xml:36]: unused

File: capa-spi-aws-log/pom.xml [hunk 10]
```diff
@@ -33,14 +33,24 @@
...
+        <sentinel.core.version>1.8.2</sentinel.core.version>
+        <http.client.version>4.5.13</http.client.version>
```
Comment (kevinten10) [inline capa-spi-aws-log/pom.xml:38]: this is all aws used

File: capa-spi-aws-log/pom.xml [hunk 11]
```diff
@@ -33,14 +33,24 @@
...
+        <netty.bom.version>4.1.68.Final</netty.bom.version>
+        <http.core.version>4.4.11</http.core.version>
```
Comment (kevinten10) [inline capa-spi-aws-log/pom.xml:40]: this is all aws used

File: capa-spi-aws-log/pom.xml [hunk 12]
```diff
@@ -88,7 +98,29 @@
...
         </dependency>
-
+        <!--http-->
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-bom</artifactId>
+            <version>${netty.bom.version}</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
```
Comment (kevinten10) [inline capa-spi-aws-log/pom.xml:108]: bom should move to parent

File: capa-spi-aws-telemetry/pom.xml [hunk 13]
```diff
@@ -82,6 +85,24 @@
             <artifactId>gson</artifactId>
             <version>${gson.version}</version>
         </dependency>
+        <!--http-->
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-bom</artifactId>
+            <version>${netty.bom.version}</version>
+            <type>pom</type>
+            <scope>import</scope>
```
Comment (kevinten10) [inline capa-spi-aws-telemetry/pom.xml:94]: bom should move to parent

**Review by kevinten10 — COMMENTED** [review]
review

**Comment by dgzpg** [comment]
@kevinten10  review 的都已经处理了~

**Pull request merged**